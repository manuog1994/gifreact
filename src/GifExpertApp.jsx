import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
    const [ categories, setCategories ] = useState([ 'One Punch' ]);

    const onAddCategory = ( newCategory ) => {
        if ( categories.find( category => category.toLowerCase() === newCategory.toLowerCase()) ) return;
        setCategories([ newCategory, ...categories ]);
        // categories.push( newCategory );
        //setCategories( [ ...categories, 'HunterXHunter' ] );
        // setCategories( cats => [ ...cats, 'HunterXHunter' ] );
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory 
                // setCategories={ setCategories } 
                onNewCategory={ onAddCategory }
            />

            {
                categories.map( (category) => (
                    <GifGrid 
                        key={ category }
                        category={ category }
                    />
                ))
            }
        </>
    )
}