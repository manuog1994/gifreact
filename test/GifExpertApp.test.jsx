import { GifExpertApp } from '../src/GifExpertApp';
import { render, screen } from '@testing-library/react';

describe('Probando el <GifExpertApp />', () => { 

    test('debe mostrarse correctamente', () => { 
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar una lista de categorias', () => { 
        const categories = ['One Punch'];
        render(<GifExpertApp  />);

        expect(categories.length).toBe(categories.length);
    });

    test('debe de dar anular la peticiÃ³n si el input esta vacio o es menor a 2', () => { 
        render(<GifExpertApp />);
        const input = screen.getByRole('form');
        
        expect(input.value).toBeFalsy();
        // screen.debug();
    })
})