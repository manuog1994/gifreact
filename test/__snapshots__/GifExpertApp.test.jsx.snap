// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el <GifExpertApp /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar..."
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
